### PR TITLE
doc(pubsub): create landing page for Doxygen

### LINKED ISSUE #4587: Write a Cloud Pub/Sub landing Doxygen page

### PR DESCRIPTION
**Pull request by coryan**
The page is a bit incomplete, we need a quickstart, and an example on
how to use mocking, and so forth. We have separate issues open for that,
and having this file makes it easier to link those examples as we add
them.

Fixes #4587

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/4623)

### CODE REVIEW

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 1]
```diff
@@ -0,0 +1,130 @@
+/*!
+
+@mainpage Cloud Pub/Sub C++ Client Library
+
+The Cloud Pusb/Sub C++ Client library offers types and functions to use Cloud
+Pub/Sub from C++11 applications.
+
+This library requires a C++11 compiler, it is supported (and tested) on multiple
```
Comment (devbww) [inline google/cloud/pubsub/doc/pubsub-main.dox:8]: s/, it/. It/

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+## Quickstart
+
+The following instructions show you how to perform basic tasks in Cloud Pub/Sub
+using the C++ client library.
+
+### Before you begin
+
+1. Select or create a Google Cloud Platform (GCP) project using the
+   [manage resource page][resource-link]. Make a note of the project id, you
```
Comment (devbww) [inline google/cloud/pubsub/doc/pubsub-main.dox:19]: s/, you/as you/

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+There are several environment variables that can be set to configure certain
+behaviors in the library.
+
+- `GOOGLE_CLOUD_CPP_ENABLE_CLOG=yes` turns on logging in the library, basically
+  the library always "logs" but the logging infrastructure has no backend to
+  actually print anything until the application sets a backend or they set this
+  environment variable.
+
+- `GOOGLE_CLOUD_CPP_ENABLE_TRACING=rpc` turns on tracing for most gRPC
+  calls, the library injects an additional Stub decorator that prints each gRPC
```
Comment (devbww) [inline google/cloud/pubsub/doc/pubsub-main.dox:66]: s/, the/. The/

File: google/cloud/pubsub/doc/pubsub-main.dox [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+  request and response.
+
+- `GOOGLE_CLOUD_CPP_ENABLE_TRACING=rpc-streams` turns on tracing for streaming
+  gRPC calls, such as `Client::Read()`, `Client::ExecuteQuery()`, and
+  `Client::ProfileQuery()`. This can produce a lot of output, so use with
```
Comment (devbww) [inline google/cloud/pubsub/doc/pubsub-main.dox:71]: Remnant Spanner names?

File: google/cloud/pubsub/samples/samples.cc [hunk 5]
```diff
@@ -142,6 +142,29 @@ void DeleteSubscription(google::cloud::pubsub::SubscriptionAdminClient client,
...
+  [](pubsub::TopicAdminClient client, std::string const& project_id) {
+    // The actual type of `topic` is
+    // google::cloud::StatusOr<google::pubsub::v1::Topic>, but
+    // we expect it'll most often be declared with auto like this.
+    for (auto const& topic : client.ListTopics(project_id)) {
+      // Use `topic` like a smart pointer; check it before dereferencing
+      if (!topic) {
+        // `topic` doesn't contain a value, so `.status()` will contain error
+        // info
+        std::cerr << topic.status();
```
Comment (devbww) [inline google/cloud/pubsub/samples/samples.cc:158]: `... << "\n`

File: google/cloud/pubsub/samples/samples.cc [hunk 6]
```diff
@@ -164,6 +187,9 @@ void AutoRun(std::vector<std::string> const& argv) {
   std::cout << "\nRunning CreateTopic() sample\n";
   CreateTopic(publisher_client, {project_id, topic_id});
 
+  std::cout << "\nRunning the StatusOr example" << std::endl;
```
Comment (devbww) [inline google/cloud/pubsub/samples/samples.cc:190]: Choose between `std::endl` and `\n` (as is used in the other "Running ..." output).

File: google/cloud/pubsub/doc/pubsub-main.dox (hunk 1 above)
Comment (coryan) [reply google/cloud/pubsub/doc/pubsub-main.dox]: Done

File: google/cloud/pubsub/doc/pubsub-main.dox (hunk 2 above)
Comment (coryan) [reply google/cloud/pubsub/doc/pubsub-main.dox]: Done

File: google/cloud/pubsub/samples/samples.cc (hunk 6 above)
Comment (coryan) [reply google/cloud/pubsub/samples/samples.cc:190]: Done

File: google/cloud/pubsub/samples/samples.cc (hunk 5 above)
Comment (coryan) [reply google/cloud/pubsub/samples/samples.cc]: Fixed

File: google/cloud/pubsub/doc/pubsub-main.dox (hunk 4 above)
Comment (coryan) [reply google/cloud/pubsub/doc/pubsub-main.dox]: Removed

File: google/cloud/pubsub/doc/pubsub-main.dox (hunk 3 above)
Comment (coryan) [reply google/cloud/pubsub/doc/pubsub-main.dox]: Done

**Pull request merged**